### DISCUSSION TITLE
Query Timeout if "too many" columns are selected

### DISCUSSION THREAD

**Discussion by johnweidauer** [mysql-net/MySqlConnector#1485] [Q&A]
I have a query that returns data, but if I include too many columns in my web application using mysqlconnector, it will timeout. I have written this query both is plain text and in a procedure, but when more than 15 columns are to be returned, timeout. Is there a column limit, or a size limit on the return? I have tried searching for this topic, but was unable to find anything. Can anyone tell me if this is a column limit or a size limit. The query in question is only returning 1 row of all columns. *This table was not created by me and is a table created by the vendor, but I do have access to make modifications to both the table and MySQL Server 5.7.

My Schema:
`CarrierKey` int(11) NOT NULL AUTO_INCREMENT,
  `code` varchar(50) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT NULL,
  `name_company` varchar(255) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT NULL,
  `address` varchar(255) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT NULL,
  `city` varchar(50) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT NULL,
  `state_code` varchar(50) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT NULL,
  `zip` varchar(50) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT NULL,
  `phone` varchar(50) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT NULL,
  `fax` varchar(50) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT NULL,
  `contact` varchar(50) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT NULL,
  `fee_schedule` varchar(50) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT NULL,
  `class` varchar(50) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT NULL,
  `fees` varchar(50) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT 'STAND',
  `edi_format` varchar(50) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT NULL,
  `code_payment` varchar(50) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT NULL,
  `code_payment2` varchar(50) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT NULL,
  `code_adjustment` varchar(50) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT NULL,
  `receiver_type` varchar(50) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT NULL,
  `code_category` varchar(50) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT NULL,
  `claim_filing` varchar(255) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT NULL,
  `payer_id_prof` varchar(50) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT NULL,
  `unique_provider_id` varchar(255) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT NULL,
  `network_status` varchar(255) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT NULL,
  `state_specific` varchar(50) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT ''

**Comment by bgrainger** [comment]
> Is there a column limit, or a size limit on the return?

There's nothing in MySqlConnector that enforces a limit; it sends the query to the server, then waits for the response to come in over the network. A timeout generally indicates that no such response was received in time.

It's hard to speculate about what might be happening but could there be some component between the database server and your application that drops packets over a certain size?
